packages/slice-machine/lib/env/manifest.ts

```diff
@@ -123,13 +72,32 @@ function handleManifest(cwd: string): ManifestInfo {
 
     return pipe(
       Models.Manifest.decode(json),
-      fold(
+      fold<t.Errors, Models.Manifest, ManifestInfo>(
         // failure handler
-        () => {
-          throw new Error();
+        (errors) => {
+          const messages = formatValidationErrors(errors, {});
+          const message = messages
+            .map((error) => "[sm.json] " + error)
+            .join("\n");
+
+          return {
+            state: ManifestState.InvalidJson,
+            message,
+            content: null,
+          };
         },
         // success handler
-        (manifest) => validate(manifest)
+        (manifest) => {
+          const endpoint = fromUrl(manifest.apiEndpoint);
+          const parsedRepo = parseDomain(endpoint);
+          const repo = extractRepo(parsedRepo);
+          return {
+            state: ManifestState.Valid,
+            message: Messages[ManifestState.Valid],
+            content: manifest,
+            repo,
+          };
+        }
       )
     );
   } catch (e) {
```

Comment:
do we still need that catch ?
Sound redundant

Reply:
it's for the call to readFileSync further up :)

Reply:
Then this can be eliminated using helpers from the Core